src/output-tx.c

```diff
@@ -138,35 +148,87 @@ static TmEcode OutputTxLog(ThreadVars *tv, Packet *p, void *thread_data, PacketQ
         if (!(AppLayerParserStateIssetFlag(f->alparser, APP_LAYER_PARSER_EOF)))
         {
             int tx_progress = AppLayerParserGetStateProgress(p->proto, alproto,
-                                                             tx, FlowGetDisruptionFlags(f, STREAM_TOSERVER));
-            if (tx_progress < tx_progress_done_value_ts) {
-                SCLogDebug("progress not far enough, not logging");
-                break;
+                tx, FlowGetDisruptionFlags(f, STREAM_TOSERVER));
+            if (tx_progress == tx_progress_done_value_ts) {
+                ts_ready = 1;
             }
 
             tx_progress = AppLayerParserGetStateProgress(p->proto, alproto,
-                                                         tx, FlowGetDisruptionFlags(f, STREAM_TOCLIENT));
-            if (tx_progress < tx_progress_done_value_tc) {
-                SCLogDebug("progress not far enough, not logging");
-                break;
+                tx, FlowGetDisruptionFlags(f, STREAM_TOCLIENT));
+            if (tx_progress == tx_progress_done_value_tc) {
+                tc_ready = 1;
             }
         }
 
+        if (!(ts_ready || tc_ready)) {
+            SCLogNotice("progress not for enough, not logging");
+            break;
+        }
+
+        /* Prevent the processing of one direction getting farther
```

Comment:
Whats the reasoning here? In for example HTTP pipelining we can have many requests before we see a response. It would make sense to log them in the order that we get them.